tests/python/pytorch/sparse/test_sparse_matrix.py

```diff
@@ -504,6 +504,66 @@ def test_range_select(create_func, shape, dense_dim, select_dim, rang):
     assert torch.allclose(A_select_to_dense, dense_select)
 
 
+@pytest.mark.parametrize(
+    "create_func", [rand_diag, rand_csr, rand_csc, rand_coo]
+)
+@pytest.mark.parametrize("sample_dim", [0, 1])
+@pytest.mark.parametrize("index", [(0, 1, 2, 3, 4), (0, 1, 3), (1, 1, 2)])
+@pytest.mark.parametrize("replace", [False, True])
+@pytest.mark.parametrize("bias", [False, True])
+def test_sample(create_func, sample_dim, index, replace, bias):
+    ctx = F.ctx()
+    shape = (5, 5)
+    sample_num = 3
+    A = create_func(shape, 10, ctx)
+    A = val_like(A, torch.abs(A.val))
+
+    index = torch.tensor(index).to(ctx)
+
+    A_sample = A.sample(sample_dim, sample_num, index, replace, bias)
+    A_dense = sparse_matrix_to_dense(A)
+    A_sample_to_dense = sparse_matrix_to_dense(A_sample)
+
+    if sample_dim == 0:
```

Comment:
nit: The nested code makes it hard to read.
One option is separate the dim 1 and 0 to 2 different test cases since the primary code is not shared anyway.

def test_sample_rowwise(create_func, index, replace, bias):

def test_sample_columnwise(create_func, index, replace, bias):